packages/embarkjs/src/utils.js

```diff
@@ -53,27 +57,90 @@ let Utils = {
 
       //toSend.estimateGas()
       //  .then(gasEstimated => {
-          //params.gas = gasEstimated + 1000;
-          return toSend.send(params, function(err, transactionHash) {
-            if (err) {
-              return callback(err);
-            }
-            hash = transactionHash;
-            hashCb(hash);
-          })
-          .on('receipt', function(receipt) {
-            if (!isContractDeploy || receipt.contractAddress) {
-              callback(null, receipt);
-            }
-          }).then(function(_contract) {
-            if (!hash) {
-              return; // Somehow we didn't get the receipt yet... Interval will catch it
-            }
-            web3.eth.getTransactionReceipt(hash, callback);
-          }).catch(callback);
-        //});
+      //params.gas = gasEstimated + 1000;
+      return toSend.send(params, function(err, transactionHash) {
+        if (err) {
+          return callback(err);
+        }
+        hash = transactionHash;
+        (hashCb || _hashCb)(hash);
+      })
+        .on('receipt', function(receipt) {
+          if (!isContractDeploy || receipt.contractAddress) {
+            callback(null, receipt);
+          }
+        }).then(function(_contract) {
+          if (!hash) {
+            return; // Somehow we didn't get the receipt yet... Interval will catch it
+          }
+          web3.eth.getTransactionReceipt(hash, callback);
+        }).catch(callback);
+      //});
     });
+
+    if (cb) { promise.then(_cb).catch(_cb); return; }
+    return promise;
   }
 };
 
 export default Utils;
+
+var getOwnPropertyDescriptors = Object.getOwnPropertyDescriptors ||
+    function getOwnPropertyDescriptors(obj) {
+      var keys = Object.keys(obj);
+      var descriptors = {};
+      for (var i = 0; i < keys.length; i++) {
```

Comment:
Would be better with `keys.forEach((key, i) => {...});`

Reply:
Possibly, it's just copied from the [util](https://github.com/defunctzombie/node-util/blob/master/util.js#L22) package, which in turn is based on the [node.js sources](https://github.com/nodejs/node/blob/3867ef02440e1f207ef4cf70367fba6028dc0064/lib/util.js#L340).

Reply:
I did a by-hand surgery on `yarn.lock` to pick up the latest version of the `util` package compatible with webpack's range for the same, per its `node-libs-browser package`. After a reboot, I was able to completely remove the inlined `callbackify` code and rely on `import`/`require`.